src/UI/DatePicker.elm

```diff
@@ -0,0 +1,477 @@
+                    String.fromInt (Date.day day.date)
+            ]
+
+
+{-| Draw the name of the day.
+-}
+drawDayName : String -> Element msg
+drawDayName day =
+    Element.el [ Element.width (fillPortion 7), Element.centerX, Element.centerY ] <|
+        Element.el [ Element.centerX, Font.size 12, Palette.gray700 |> Palette.toFontColor ]
+            (Element.text <| day)
+
+
+{-| Draw the header of the calendar.
+-}
+header : RenderConfig -> (Msg -> msg) -> Property msg -> Element msg
+header renderConfig externalMsg model =
+    let
+        label =
+            Date.format "MMMM" model.current ++ " " ++ (String.fromInt <| Date.year model.current)
+
+        iconSize =
+            if RenderConfig.isMobile renderConfig then
+                Size.medium
+
+            else
+                Size.small
+    in
+    Element.row [ Element.width fill ]
+        [ Button.fromIcon (Icon.chevronLeft "Previous Month")
+            |> Button.cmd (externalMsg PreviousMonth) Button.clear
+            |> Button.withSize iconSize
+            |> Button.renderElement renderConfig
+            |> Element.el [ alignLeft ]
+        , label
+            |> Text.subtitle1
+            |> Text.withColor Palette.gray800
+            |> Text.renderElement renderConfig
+            |> Element.el [ Element.centerX ]
+        , Button.fromIcon (Icon.chevronRight "Next Month")
+            |> Button.cmd (externalMsg NextMonth) Button.clear
+            |> Button.withSize iconSize
+            |> Button.renderElement renderConfig
+            |> Element.el [ alignRight ]
+        ]
+
+
+weekHeader : RenderConfig -> List String
+weekHeader _ =
+    [ "Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun" ]
+
+
+{-| Numbers of the days to show in the calendar
+-}
+daysInTheView : Int
+daysInTheView =
+    42
+
+
+{-| Return the week-day number of the 1th of the month
+-}
+weekDayOfFirstDay : Date -> Int
+weekDayOfFirstDay date =
+    let
+        firstDay =
+            Date.fromCalendarDate (Date.year date) (Date.month date) 1
+    in
+    Date.weekdayNumber firstDay
+
+
+{-| Return the days to show from the previous month
+-}
+dateDaysFromPreviousMonth : Date -> List DayItem
+dateDaysFromPreviousMonth date =
+    let
+        firstWeekDayMonth =
+            weekDayOfFirstDay date
+    in
+    if firstWeekDayMonth == 1 then
+        []
+
+    else
+        let
+            year =
+                Date.year date
+
+            month =
+                Date.month date
+
+            first =
+                Date.fromCalendarDate year month 1
+
+            start =
+                Date.add Days (-firstWeekDayMonth + 1) first
+
+            until =
+                Date.add Days -1 first
+
+            dates =
+                Date.range Day 1 start until ++ List.singleton until
+        in
+        List.map
+            (\day ->
+                { date = day
+                , isCurrentMonth = False
+                }
+            )
+            dates
```

Comment:
ugly but nothing wrong with it

Reply:
I think I can rewrite it using a fold and [subtracting days](https://package.elm-lang.org/packages/justinmimbs/date/4.0.0/Date#add)